# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/misc/vice/hotfix-binutils-gcc.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Hotfix needed for new binutils (bug is in gcc)
(see http://gcc.gnu.org/bugzilla/show_bug.cgi?id=16625)

--- ./src/resid/wave.h.orig	2004-08-13 16:25:59.000000000 +0200
+++ ./src/resid/wave.h	2004-08-13 16:28:16.000000000 +0200
@@ -462,7 +462,6 @@
   // waveform output; however a switch with inline functions is faster.
 
   switch (waveform) {
-  default:
   case 0x0:
     return output____();
   case 0x1:
@@ -471,6 +470,8 @@
     return output__S_();
   case 0x3:
     return output__ST();
+  }
+  switch (waveform) {
   case 0x4:
     return output_P__();
   case 0x5:
@@ -479,6 +480,8 @@
     return output_PS_();
   case 0x7:
     return output_PST();
+  }
+  switch (waveform) {
   case 0x8:
     return outputN___();
   case 0x9:
@@ -487,6 +490,8 @@
     return outputN_S_();
   case 0xb:
     return outputN_ST();
+  }
+  switch (waveform) {
   case 0xc:
     return outputNP__();
   case 0xd:
@@ -496,6 +501,7 @@
   case 0xf:
     return outputNPST();
   }
+  return output____();
 }
 
 #endif // RESID_INLINING || defined(__WAVE_CC__)
